tinyDA/diagnostics.py

```diff
@@ -207,3 +207,48 @@ def get_samples(chain, attribute="parameters", level="fine", burnin=0):
 
     # return the samples.
     return samples
+
+def DA_estimator(chain, attribute="qoi", variable="x0", burnin=0):
+    '''Computes the unbiased Monte-Carlo estimator for Delayed Acceptance (two-level),
+    as derived in Lykkegaard et al. 2023.
+
+    Parameters
+    ----------
+    chain : dict
+        A dict as returned by tinyDA.sample, containing chain information
+        and lists of tinyDA.Link instances.
+    attribute : str, optional
+        Which link attribute ('parameters', 'model_output', 'qoi' or 'stats')
+        to extract. The default is 'parameters'.
+    variable : str, optional
+        Which variable of the posterior or qoi to marginalize over.
+    burnin : int, optional
+        The burnin length. The default is 0.
+
+    Returns
+    ----------
+    float
+        Outpu of the estimator computation.
+
+    '''
+
+    inferencedata_coarse = to_inference_data(chain, level='coarse', burnin=burnin)
+    inferencedata_coarse_promoted = to_inference_data(chain, level='promoted_coarse', burnin=burnin)
+    inferencedata_fine = to_inference_data(chain, level='fine', burnin=burnin)
+
+    qoi1 = 0
+    qoi2 = 0
+
+    if attribute=="qoi":
+        qoi1 = inferencedata_coarse_promoted.qoi
+        qoi2 = inferencedata_fine.qoi
+    if attribute=="posterior":
+        qoi1 = inferencedata_coarse_promoted.posterior
+        qoi2 = inferencedata_fine.posterior
+
+    qoi_fine_corr = qoi1 - qoi2
```

Comment:
I prefer more explanatory names. Can we rename them
`{"qoi1": "values_coarse_promoted",
  "qoi2": "values_fine",
  "qoi_fine_corr": "values_difference",
  "mean_fine_corr": "mean_difference"}`